pkg/minikube/node/start.go

```diff
@@ -156,9 +156,23 @@ func Start(starter Starter, apiServer bool) (*kubeconfig.Settings, error) {
 			return nil, errors.Wrap(err, "Updating node")
 		}
 
-		cpBs, err := cluster.Bootstrapper(starter.MachineAPI, viper.GetString(cmdcfg.Bootstrapper), *starter.Cfg, starter.Runner)
+		// Make sure to use the command runner for the control plane to generate the join token
+		cp, err := config.PrimaryControlPlane(starter.Cfg)
```

Comment:
nit: this could be in same block
if cp,err:= ... ; err != nil {
}

Reply:
it can't, cp needs to be used in the following block so we need to keep it in scope.